whois-update/src/main/java/net/ripe/db/whois/update/sso/SsoTranslator.java

```diff
@@ -29,12 +27,12 @@ public void populateCacheAuthToUsername(final UpdateContext updateContext, final
         SsoHelper.translateAuth(rpslObject, new AuthTranslator() {
             @Override
             @CheckForNull
-            public RpslAttribute translate(final String authType, final String authToken, final RpslAttribute originalAttribute) {
+            public RpslAttribute translate(final String authType, final String uuid, final RpslAttribute originalAttribute) {
                 if (authType.equals("SSO")) {
-                    if (!updateContext.hasSsoTranslationResult(authToken)) {
+                    if (!updateContext.getSsoTranslation().containsUuid(uuid)) {
                         try {
-                            final String username = authServiceClient.getUsername(authToken);
-                            updateContext.addSsoTranslationResult(authToken, username);
+                            final String username = authServiceClient.getUsername(uuid);
+                            updateContext.getSsoTranslation().put(username, uuid);
```

Comment:
Do we need the extra `contains()` checks within the `put()` method? Is it necessary to prevent overwriting existing values? It seems excessive if there's already a contains() call first.

Reply:
inside the put we are checking both the username and the uuid
here there is a contains to directly avoid doing rest of the code (the call to auth included) if that uuid already exist.
SO we have a first contains to check just the UUID because is the info that we have to avoid doing unnecessary code, and once we have both information username and uuid we check both

Reply:
so, you are right we can leave just the contains inside the put() this will work.
However, there is the possibility that this code will be execute when the UUID already exist in the map